internals/daemon/api_files.go

```diff
@@ -501,17 +501,31 @@ func writeFile(item writeFilesItem, source io.Reader) error {
 
 func mkdirAllUserGroup(path string, perm os.FileMode, uid, gid *int) error {
 	if uid != nil && gid != nil {
-		return mkdirAllChown(path, perm, sys.UserID(*uid), sys.GroupID(*gid))
+		return mkdir(path, perm, &osutil.MkdirOptions{
+			MakeParents: true,
+			ExistOK:     true,
+			Chown:       true,
+			UserID:      sys.UserID(*uid),
+			GroupID:     sys.GroupID(*gid),
+		})
 	} else {
-		return mkdirAllChown(path, perm, osutil.NoChown, osutil.NoChown)
+		return mkdir(path, perm, &osutil.MkdirOptions{
+			MakeParents: true,
+			ExistOK:     true,
+		})
 	}
 }
 
 func mkdirUserGroup(path string, perm os.FileMode, uid, gid *int) error {
 	if uid != nil && gid != nil {
-		return mkdirChown(path, perm, sys.UserID(*uid), sys.GroupID(*gid))
+		return mkdir(path, perm, &osutil.MkdirOptions{
+			ExistOK: true,
```

Comment:
Just to double-check: does the old `osutil.MkdirChown` have "exist-ok" semantics? If so, shouldn't we use `ExistOK` on line 528 in the `else` clause as well?

Reply:
I don't think there is, see [here](https://github.com/canonical/pebble/blob/master/internals/osutil/mkdirallchown.go#L68). Only the original `osutil.MkdirAllChown` has this logic [here](https://github.com/canonical/pebble/blob/master/internals/osutil/mkdirallchown.go#L45).

Reply:
But in that case it looks like the original mkdirChown doesn't have exist-ok semantics at all, so neither of these should, right?

Reply:
Yes (updated the previous comment).